src/main/scala/com.snowplowanalytics.snowplow.analytics.scalasdk/ParsingError.scala

```diff
@@ -0,0 +1,36 @@
+package com.snowplowanalytics.snowplow.analytics.scalasdk
+
+/**
+  * Represents error during parsing TSV event
+  */
+sealed trait ParsingError
+
+object ParsingError {
+
+  /**
+    * Represents error which given line is not a TSV
+    */
+  case object NonTSVPayload extends ParsingError
+
+  /**
+    * Represents error which number of given columns is not equal
+    * to number of expected columns
+    * @param expectedNum expected number of columns
+    * @param gotNum number of given columns
+    */
+  case class ColumnNumberMismatch(expectedNum: Int, gotNum: Int) extends ParsingError
+
+  /**
+    * Represents cases where value in a field is not valid
+    * e.g invalid timestamp, invalid UUID
+    * @param key key of field
+    * @param value value of the field
+    */
+  case class InvalidValue(key: String, value: String) extends ParsingError
+
+  /**
+    * Represents unexpected errors
+    * @param error error message
+    */
+  case class UnexpectedError(error: String) extends ParsingError
```

Comment:
Are we sure this is really unexpected error and can never happen?

Reply:
I've used this error in [here](https://github.com/snowplow/snowplow-scala-analytics-sdk/pull/83/commits/89af6572151518b421cc0e80fce8f0b5a7e6c724#diff-cf028adf3436d2dea6cace595dcb0ac3R48) and it seems to getting to places where this error is thrown is impossible due to types of the implicit RowDecoder instances. If I am missing anything, please let me know.

Reply:
Nope, I also think so. Just wanted to clarify.